pilot/pkg/networking/core/listener_builder.go

```diff
@@ -443,9 +443,5 @@ func appendMxFilter(httpOpts *httpListenerOpts, filters []*hcm.HttpFilter) []*hc
 	if httpOpts.class == istionetworking.ListenerClassSidecarInbound {
 		return append(filters, xdsfilters.SidecarInboundMetadataFilter)
 	}
-
-	if httpOpts.skipIstioMXHeaders {
```

Comment:
we seem to lose this? `skipIstioMXHeaders`

Reply:
Yes, will add it later.